cluster/kubernetes/sync.go

```diff
@@ -198,9 +199,20 @@ func (c *Cluster) getAllowedResourcesBySelector(selector string) (map[string]*ku
 		listOptions.LabelSelector = selector
 	}
 
-	resources, err := c.client.discoveryClient.ServerResources()
+	_, resources, err := c.client.discoveryClient.ServerGroupsAndResources()
 	if err != nil {
-		return nil, err
+		discErr, ok := err.(*discovery.ErrGroupDiscoveryFailed)
+		if !ok {
+			return nil, err
+		}
+		for gv, e := range discErr.Groups {
+			// Tolerate empty GroupVersions due to e.g. misconfigured custom metrics
+			if e.Error() != fmt.Sprintf("Got empty response for: %v", gv) {
+				return nil, err
+			}
+		}
+		c.logger.Log("warn", err,
+			"impact", "ignoring error, please check your cluster configuration, GroupVersions should not be empty")
```

Comment:
I am on the fence about this. It feels like telling users off without any real advantage.

I think it's probably fine to be silent about it.

Reply:
Maybe just log it once? Or if it's rare, and there's some characteristic symptom of it failing, we can be silent here and add a troubleshooting doc entry instead.

Reply:
As we talked about reducing the log clutter a while ago and the reports about this problem thus far have been Datadog metric users (so _I think_ the problem is very niche), I agree with you @fons and it is probably fine to be silent about it.

Reply:
I will silence it then